include/zephyr/drivers/misc/devmux/devmux.h

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2023, Meta
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Public APIs for the Device Multiplexer driver
+ */
+
+#ifndef INCLUDE_ZEPHYR_DRIVERS_MISC_DEVMUX_H_
+#define INCLUDE_ZEPHYR_DRIVERS_MISC_DEVMUX_H_
+
+#include <stdint.h>
+
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Devmux Driver APIs
+ * @defgroup demux_interface Devmux Driver APIs
+ * @ingroup misc_interfaces
+ *
+ * @details
+ * Devmux operates as a device multiplexer, forwarding the characteristics of
+ * the selected device.
+ *
+ * ```
+ *            +----------+                            +----------+
+ *            |  devmux  |                            |  devmux  |
+ *            |          |                            |          |
+ * dev0       |          |                 dev0       |          |
+ * +---------->   \      |                 +---------->          |
+ *            |    \     |                            |          |
+ * dev1       |     \    |       dev0      dev1       |          |       dev2
+ * +---------->      O   +---------->      +---------->      O   +---------->
+ *            |          |                            |     /    |
+ * dev2       |          |                 dev2       |    /     |
+ * +---------->          |                 +---------->   /      |
+ *            |          |                            |          |
```

Comment:
drop a line? 3 up, 4 down... the asymmetry is killing me

Reply:
Not sure what you mean

Reply:
ah that's fine, the `devmux` label is centered so I guess you have to draw around that... ascii art is hard!

Reply:
Fixed!